docs/using-the-compiler.rst

```diff
@@ -0,0 +1,249 @@
+        optimizer: {enabled: true, runs: 500},
+        // if given, only compiles this contract, can also be an array. If only a contract name is given, tries to find it if unique.
+        compilationTarget: "myFile.sol:MyContract",
+        // addresses of the libraries. If not all libraries are given here, it can result in unlinked objects whose output data is different
+        libraries: {
+          "def:MyLib": "0x123123..."
+        },
+        // The following can be used to restrict the fields the compiler will output.
+        // (axic)
+        outputSelection: [
+            "abi", "evm.assembly", "evm.bytecode", ..., "why3", "ewasm.wasm"
+        ]
+        outputSelection: {
+        abi,asm,ast,bin,bin-runtime,clone-bin,devdoc,interface,opcodes,srcmap,srcmap-runtime,userdoc
+
+ --ast                 AST of all source files.
+  --ast-json            AST of all source files in JSON format.
+  --asm                 EVM assembly of the contracts.
+  --asm-json            EVM assembly of the contracts in JSON format.
+  --opcodes             Opcodes of the contracts.
+  --bin                 Binary of the contracts in hex.
+  --bin-runtime         Binary of the runtime part of the contracts in hex.
+  --clone-bin           Binary of the clone contracts in hex.
+  --abi                 ABI specification of the contracts.
+  --interface           Solidity interface of the contracts.
+  --hashes              Function signature hashes of the contracts.
+  --userdoc             Natspec user documentation of all contracts.
+  --devdoc              Natspec developer documentation of all contracts.
+  --formal              Translated source suitable for formal analysis.
+
+          // to be defined
+        }
+      }
+    }
+
+
+Regular Output
+--------------
+
+
+    {
+      errors: ["error1", "error2"], // we might structure them
+      errors: [
+          {
+              // (axic)
+              file: "sourceFile.sol", // optional?
+              contract: "contractName", // optional
+              line: 100, // optional - currently, we always have a byte range in the source file
+              // Errors/warnings originate in several components, most of them are not
+              // backend-specific. Currently, why3 errors are part of the why3 output.
+              // I think it is better to put code-generator-specific errors into the code-generator output
+              // area, and warnings and errors that are code-generator-agnostic into this general area,
+              // so that it is easier to determine whether some source code is invalid or only
+              // triggers errors/warnings in some backend that might only implement some part of solidity.
+              type: "evm" or "why3" or "ewasm" // maybe a better field name would be needed
+              severity: "warning" or "error" // mandatory
+              message: "Invalid keyword" // mandatory
+          }
+      ]
+      contracts: {
```

Comment:
This contains all the compiled outputs. It can be limited the `compilationTarget` setting.